aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java

```diff
@@ -97,12 +99,13 @@ public void initialize(Map<String, String> properties) {
 
     clientBuilder.credentialsProvider(
         StsAssumeRoleCredentialsProvider.builder()
-            .stsClient(StsClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build())
+            .stsClient(StsClient.builder().httpClientBuilder(
```

Comment:
I think we should have a private method for `sts()` similar to what you have for `lakeformation()`, it makes the code more clear about all the AWS clients